### PR TITLE
bpo-31680: Add curses.ncurses_version.

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
https://bugs.python.org/issue31680

### CODE REVIEW

File: Lib/test/test_curses.py [hunk 1]
```diff
@@ -447,6 +446,22 @@ def test_update_lines_cols(self):
...
+        self.assertIsInstance(v[0], int)
+        self.assertIsInstance(v[1], int)
+        self.assertIsInstance(v[2], int)
+        self.assertIsInstance(v.major, int)
+        self.assertIsInstance(v.minor, int)
+        self.assertIsInstance(v.patch, int)
+        self.assertEqual(v[0], v.major)
+        self.assertEqual(v[1], v.minor)
+        self.assertEqual(v[2], v.patch)
+        self.assertTrue(v > (0, 0, 0))
```
Comment (vstinner) [inline Lib/test/test_curses.py]: I would prefer to check that each field is > 0.

```
self.assertGreater(v.major, 0)
self.assertGreater(v.minor, 0)
self.assertGreater(v.patch, 0)
```

``> (0, 0, 0)`` test is not enough:

```
>>> (1, -2, 3) > (0, 0, 0)
True
```

File: Doc/library/curses.rst [hunk 2]
```diff
@@ -1279,6 +1279,18 @@ The :mod:`curses` module defines the following data members:
    A bytes object representing the current version of the module.  Also available as
    :const:`__version__`.
 
+
+.. data:: ncurses_version
+
+   A named tuple containing the three components of the ncurses library
+   version: *major*, *minor*, and *patch*.  All values are integers.  The
+   components can also be accessed by name,  so ``curses.ncurses_version[0]``
+   is equivalent to ``curses.ncurses_version.major`` and so on.  Available
```
Comment (vstinner) [inline Doc/library/curses.rst]: Please write "Available ..." in a new paragraph, to have a similar style than "Availability: ..." used in the os documentation.

**Review by vstinner — COMMENTED** [review]
I don't understand when curses.ncurses_version is not available: it seems to always be defined in the C code?

I like tuple-like API of curses.ncurses_version, it's easy to write curses.ncurses_version >= (5, 7)!

File: Lib/test/test_curses.py (hunk 1 above)
Comment (serhiy-storchaka) [reply Lib/test/test_curses.py]: v.minor is 0 in the actual version.

File: Lib/test/test_curses.py [hunk 3]
```diff
@@ -447,6 +446,23 @@ def test_update_lines_cols(self):
...
+        self.assertIsInstance(v[2], int)
+        self.assertIsInstance(v.major, int)
+        self.assertIsInstance(v.minor, int)
+        self.assertIsInstance(v.patch, int)
+        self.assertEqual(v[0], v.major)
+        self.assertEqual(v[1], v.minor)
+        self.assertEqual(v[2], v.patch)
+        self.assertGreaterEqual(v.major, 0)
+        self.assertGreaterEqual(v.minor, 0)
+        self.assertGreaterEqual(v.patch, 0)
```
Comment (vstinner) [inline Lib/test/test_curses.py:474]: Perfect, thank you :-)

File: Modules/_cursesmodule.c [hunk 4]
```diff
@@ -3361,6 +3414,30 @@ PyInit__curses(void)
     PyDict_SetItemString(d, "__version__", v);
     Py_DECREF(v);
 
+#ifdef NCURSES_VERSION
```
Comment (vstinner) [inline Modules/_cursesmodule.c:4482]: Ah ok, now it's more explicit :-)

File: Misc/NEWS.d/next/Library/2017-11-01-15-44-48.bpo-31680.yO6oSC.rst [hunk 5]
```diff
@@ -0,0 +1 @@
+Added :data:`curses.ncurses_version`.
```
Comment (vstinner) [inline Misc/NEWS.d/next/Library/2017-11-01-15-44-48.bpo-31680.yO6oSC.rst:1]: You may add it to Doc/whatsnew/3.7.rst as well. It's up to you.

**Review by vstinner — APPROVED** [review]
LGTM. I just proposed to mention the new feature in What's New in Python 3.7.

**Comment by taleinat** [comment]
Ping, @serhiy-storchaka? This looks good to go!

**Comment by vstinner** [comment]
Yep, this PR still LGTM.

**Comment by serhiy-storchaka** [comment]
I wanted first to add similar features in other modules which are interfaces to external libraries: zlib, expat, sqlite, tkinter, etc. This may lead to changing this PR for unifying with other versions.

**Comment by vstinner** [comment]
> I wanted first to add similar features in other modules which are interfaces to external libraries: zlib, expat, sqlite, tkinter, etc. This may lead to changing this PR for unifying with other versions.

The API is fine. I don't see why the *API* would change. If you would like to refactor the implementation, that can be done later. I suggest to merge this PR.

**Pull request merged**